salt/states/file.py

```diff
@@ -334,6 +334,15 @@ def run():
 }
 
 
+def _http_ftp_check(source):
+    """
+    Check if source or sources is http or ftp.
+    """
+    if isinstance(source, str):
+        return source.lower().startswith(("http:", "ftp:"))
+    return all([s.lower().startswith(("http:", "ftp:")) for s in source])
```

Comment:
So, if one of the sources in the list is https we ignore the source hash for all?